grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java

```diff
@@ -71,17 +73,28 @@ public void setupSession(){
     }
 
     @Test
-    public void whenSingleQueryLoadedAndTaskCompletionFunctionThrowsError_ErrorIsLogged(){
+    public void whenSingleQueryLoadedAndTaskCompletionFunctionThrowsError_ErrorIsLogged() throws InterruptedException {
+        CountDownLatch callbackCompleted = new CountDownLatch(1);
+
         // Create a BatchMutatorClient with a callback that will fail
         BatchMutatorClient loader = loader();
-        loader.setTaskCompletionConsumer((json) -> assertTrue("Testing Log failure",false));
+        loader.setTaskCompletionConsumer((json) -> {
+            try {
+                fail("Testing failures appear in logs");
```

Comment:
Why is that a failure? What happens when a Assert.fail is called asynchronously? Does it actually fail the test?